src/app/data_file_manager/data_file_manager.cpp

```diff
@@ -19,6 +21,52 @@ void DataFileManager::promptReceptionist(clinic_data::Workers::Receptionist* new
     new_receptionist->set_gender(receptionist_data->getGender());
 }
 
+void DataFileManager::promptVisit(clinic_data::VisitsData::Visit* new_visit, const std::shared_ptr<Visit>& visit_data)
+{
+    if (!visit_data->getDoctor().expired())
+    {
+        clinic_data::VisitsData::Doctor* doctor = new_visit->mutable_doctor();
+        doctor->set_name(visit_data->getDoctor().lock()->getName());
+        doctor->set_surname(visit_data->getDoctor().lock()->getSurname());
+        doctor->set_pesel(visit_data->getDoctor().lock()->getPesel());
+        doctor->set_gender(visit_data->getDoctor().lock()->getGender());
+    }
```

Comment:
two many locks like visit_data->getDoctor().lock() and comparison whith expired combined.
I suggest locking the visit_data->getDoctor() once and using this locked_ptr:
```
auto visit_doctor = visit_data->getDoctor().lock();
if (visit_doctor)
{
    clinic_data::VisitsData::Doctor* doctor = new_visit->mutable_doctor();
    doctor->set_name(visit_doctor->getName());
    doctor->set_surname(visit_doctor->getSurname());
    doctor->set_pesel(visit_doctor->getPesel());
    doctor->set_gender(visit_doctor->getGender());
}
```

Reply:
Nice idea